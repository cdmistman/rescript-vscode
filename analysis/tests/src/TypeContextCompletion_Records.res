type someRecord = {somethingElse: int, whatIsThis: bool}

type anotherRecord = {something: someRecord, anotherThing: option<someRecord>, thirdThing: string}

let someVal = {
  something: {somethingElse: 123, whatIsThis: false},
  anotherThing: None,
  thirdThing: "test",
}

// let value: anotherRecord = {
//   anotherThing: None,
//   some
//       ^com

// let {another} = someVal
//             ^com
