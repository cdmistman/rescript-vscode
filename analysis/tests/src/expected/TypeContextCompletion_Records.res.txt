Complete src/TypeContextCompletion_Records.res 16:15
posCursor:[16:15] posNoWhite:[16:14] Found pattern:[16:7->16:16]
posCursor:[16:15] posNoWhite:[16:14] Found pattern:[16:8->16:15]
Completable: CtypedContext(Value[someVal])=another pattern: [RecordField:another] seenIdents: []
[]

Complete src/TypeContextCompletion_Records.res 19:23
posCursor:[19:23] posNoWhite:[19:22] Found pattern:[19:7->19:24]
posCursor:[19:23] posNoWhite:[19:22] Found pattern:[19:22->19:23]
Completable: CtypedContext(Value[someVal])=s pattern: [RecordField:s] seenIdents: [anotherThing]
[]

Complete src/TypeContextCompletion_Records.res 22:8
posCursor:[22:8] posNoWhite:[22:7] Found pattern:[22:7->22:9]
Completable: CtypedContext(Value[someVal])="" pattern: [] seenIdents: []
[{
    "label": "something",
    "kind": 5,
    "tags": [],
    "detail": "something: someRecord\n\ntype anotherRecord = {\n  something: someRecord,\n  anotherThing: option<someRecord>,\n  thirdThing: string,\n}",
    "documentation": null
  }, {
    "label": "anotherThing",
    "kind": 5,
    "tags": [],
    "detail": "anotherThing: option<someRecord>\n\ntype anotherRecord = {\n  something: someRecord,\n  anotherThing: option<someRecord>,\n  thirdThing: string,\n}",
    "documentation": null
  }, {
    "label": "thirdThing",
    "kind": 5,
    "tags": [],
    "detail": "thirdThing: string\n\ntype anotherRecord = {\n  something: someRecord,\n  anotherThing: option<someRecord>,\n  thirdThing: string,\n}",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 31:8
posCursor:[31:8] posNoWhite:[31:7] Found pattern:[31:7->31:9]
Completable: CtypedContext(Value[getSomeVal](~irrelevant))="" pattern: [] seenIdents: []
[{
    "label": "something",
    "kind": 5,
    "tags": [],
    "detail": "something: someRecord\n\ntype anotherRecord = {\n  something: someRecord,\n  anotherThing: option<someRecord>,\n  thirdThing: string,\n}",
    "documentation": null
  }, {
    "label": "anotherThing",
    "kind": 5,
    "tags": [],
    "detail": "anotherThing: option<someRecord>\n\ntype anotherRecord = {\n  something: someRecord,\n  anotherThing: option<someRecord>,\n  thirdThing: string,\n}",
    "documentation": null
  }, {
    "label": "thirdThing",
    "kind": 5,
    "tags": [],
    "detail": "thirdThing: string\n\ntype anotherRecord = {\n  something: someRecord,\n  anotherThing: option<someRecord>,\n  thirdThing: string,\n}",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 35:20
posCursor:[35:20] posNoWhite:[35:19] Found pattern:[35:7->35:22]
posCursor:[35:20] posNoWhite:[35:19] Found pattern:[35:19->35:21]
Completable: CtypedContext(Value[someVal])="" pattern: [RecordField:something] seenIdents: []
[{
    "label": "somethingElse",
    "kind": 5,
    "tags": [],
    "detail": "somethingElse: int\n\ntype someRecord = {\n  somethingElse: int,\n  whatIsThis: bool,\n  anotherLevel: anotherLevel,\n}",
    "documentation": null
  }, {
    "label": "whatIsThis",
    "kind": 5,
    "tags": [],
    "detail": "whatIsThis: bool\n\ntype someRecord = {\n  somethingElse: int,\n  whatIsThis: bool,\n  anotherLevel: anotherLevel,\n}",
    "documentation": null
  }, {
    "label": "anotherLevel",
    "kind": 5,
    "tags": [],
    "detail": "anotherLevel: anotherLevel\n\ntype someRecord = {\n  somethingElse: int,\n  whatIsThis: bool,\n  anotherLevel: anotherLevel,\n}",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 38:47
posCursor:[38:47] posNoWhite:[38:46] Found pattern:[38:7->38:50]
posCursor:[38:47] posNoWhite:[38:46] Found pattern:[38:19->38:49]
posCursor:[38:47] posNoWhite:[38:46] Found pattern:[38:46->38:48]
Completable: CtypedContext(Value[someVal])="" pattern: [RecordField:something, RecordField:anotherLevel] seenIdents: [whatIsThis]
[{
    "label": "level",
    "kind": 5,
    "tags": [],
    "detail": "level: int\n\ntype anotherLevel = {level: int, someOtherArg: bool}",
    "documentation": null
  }, {
    "label": "someOtherArg",
    "kind": 5,
    "tags": [],
    "detail": "someOtherArg: bool\n\ntype anotherLevel = {level: int, someOtherArg: bool}",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 41:48
posCursor:[41:48] posNoWhite:[41:47] Found pattern:[41:7->41:51]
posCursor:[41:48] posNoWhite:[41:47] Found pattern:[41:19->41:50]
posCursor:[41:48] posNoWhite:[41:47] Found pattern:[41:46->41:49]
posCursor:[41:48] posNoWhite:[41:47] Found pattern:[41:47->41:48]
Completable: CtypedContext(Value[someVal])=l pattern: [RecordField:something, RecordField:anotherLevel, RecordField:l] seenIdents: []
[]

Complete src/TypeContextCompletion_Records.res 44:47
posCursor:[44:47] posNoWhite:[44:44] Found pattern:[44:7->44:49]
posCursor:[44:47] posNoWhite:[44:44] Found pattern:[44:19->44:48]
[]

Complete src/TypeContextCompletion_Records.res 47:31
posCursor:[47:31] posNoWhite:[47:30] Found pattern:[47:7->47:47]
posCursor:[47:31] posNoWhite:[47:30] Found pattern:[47:19->47:46]
[]

Complete src/TypeContextCompletion_Records.res 50:44
posCursor:[50:44] posNoWhite:[50:43] Found expr:[50:3->50:54]
posCursor:[50:44] posNoWhite:[50:43] Found pattern:[50:22->50:46]
posCursor:[50:44] posNoWhite:[50:43] Found pattern:[50:42->50:45]
Completable: CtypedContext(Value[someVal])=som pattern: [RecordField:som] seenIdents: [thirdThing]
[]

Complete src/TypeContextCompletion_Records.res 53:52
posCursor:[53:52] posNoWhite:[53:51] Found expr:[53:3->53:61]
posCursor:[53:52] posNoWhite:[53:51] Found pattern:[53:51->53:53]
Completable: CtypedContext(Value[someVal])="" pattern: [] seenIdents: []
[{
    "label": "something",
    "kind": 5,
    "tags": [],
    "detail": "something: someRecord\n\ntype anotherRecord = {\n  something: someRecord,\n  anotherThing: option<someRecord>,\n  thirdThing: string,\n}",
    "documentation": null
  }, {
    "label": "anotherThing",
    "kind": 5,
    "tags": [],
    "detail": "anotherThing: option<someRecord>\n\ntype anotherRecord = {\n  something: someRecord,\n  anotherThing: option<someRecord>,\n  thirdThing: string,\n}",
    "documentation": null
  }, {
    "label": "thirdThing",
    "kind": 5,
    "tags": [],
    "detail": "thirdThing: string\n\ntype anotherRecord = {\n  something: someRecord,\n  anotherThing: option<someRecord>,\n  thirdThing: string,\n}",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 56:89
posCursor:[56:89] posNoWhite:[56:88] Found expr:[56:3->56:100]
posCursor:[56:89] posNoWhite:[56:88] Found pattern:[56:51->56:92]
posCursor:[56:89] posNoWhite:[56:88] Found pattern:[56:63->56:91]
[]

Complete src/TypeContextCompletion_Records.res 80:48
posCursor:[80:48] posNoWhite:[80:47] Found expr:[80:3->80:57]
posCursor:[80:48] posNoWhite:[80:47] Found pattern:[80:29->80:49]
posCursor:[80:48] posNoWhite:[80:47] Found pattern:[80:41->80:48]
posCursor:[80:48] posNoWhite:[80:47] Found pattern:[80:47->80:48]
Ppat_construct T:[80:47->80:48]
Completable: CtypedContext(Value[someOtherValue])=T pattern: [RecordField:something] seenIdents: []
[{
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three(_)",
    "kind": 4,
    "tags": [],
    "detail": "Three(int)\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 83:48
posCursor:[83:48] posNoWhite:[83:47] Found expr:[83:3->83:58]
posCursor:[83:48] posNoWhite:[83:47] Found pattern:[83:29->83:50]
posCursor:[83:48] posNoWhite:[83:47] Found pattern:[83:42->83:49]
Ppat_construct Some:[83:42->83:46]
posCursor:[83:48] posNoWhite:[83:47] Found pattern:[83:47->83:48]
Ppat_construct T:[83:47->83:48]
Completable: CtypedContext(Value[someOtherValue])=T pattern: [RecordField:otherThing, Variant:Some(0)] seenIdents: []
[{
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three(_)",
    "kind": 4,
    "tags": [],
    "detail": "Three(int)\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 86:47
posCursor:[86:47] posNoWhite:[86:46] Found expr:[86:3->86:57]
posCursor:[86:47] posNoWhite:[86:46] Found pattern:[86:29->86:49]
posCursor:[86:47] posNoWhite:[86:46] Found pattern:[86:42->86:48]
posCursor:[86:47] posNoWhite:[86:46] Found pattern:[86:46->86:47]
Ppat_construct T:[86:46->86:47]
Completable: CtypedContext(Value[someOtherValue])=T pattern: [RecordField:thirdStuff, Tuple:1] seenIdents: []
[{
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three(_)",
    "kind": 4,
    "tags": [],
    "detail": "Three(int)\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 89:51
posCursor:[89:51] posNoWhite:[89:50] Found expr:[89:3->89:62]
posCursor:[89:51] posNoWhite:[89:50] Found pattern:[89:29->89:54]
posCursor:[89:51] posNoWhite:[89:50] Found pattern:[89:42->89:53]
posCursor:[89:51] posNoWhite:[89:50] Found pattern:[89:46->89:52]
Ppat_construct Four:[89:46->89:50]
posCursor:[89:51] posNoWhite:[89:50] Found pattern:[89:50->89:52]
Ppat_construct ():[89:50->89:52]
Completable: CtypedContext(Value[someOtherValue])="" pattern: [RecordField:thirdStuff, Tuple:1, Variant:Four(0)] seenIdents: []
[{
    "label": "TwentyFive",
    "kind": 4,
    "tags": [],
    "detail": "TwentyFive\n\n",
    "documentation": null
  }, {
    "label": "SixtyTwo",
    "kind": 4,
    "tags": [],
    "detail": "SixtyTwo\n\n",
    "documentation": null
  }, {
    "label": "someOtherValue",
    "kind": 12,
    "tags": [],
    "detail": "someRecordWithVariant",
    "documentation": null
  }, {
    "label": "getSomeVal",
    "kind": 12,
    "tags": [],
    "detail": "(~irrelevant: int) => anotherRecord",
    "documentation": null
  }, {
    "label": "someVal",
    "kind": 12,
    "tags": [],
    "detail": "anotherRecord",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 92:55
posCursor:[92:55] posNoWhite:[92:54] Found expr:[92:3->92:66]
posCursor:[92:55] posNoWhite:[92:54] Found pattern:[92:29->92:58]
posCursor:[92:55] posNoWhite:[92:54] Found pattern:[92:42->92:57]
posCursor:[92:55] posNoWhite:[92:54] Found pattern:[92:46->92:56]
Ppat_construct Five:[92:46->92:50]
posCursor:[92:55] posNoWhite:[92:54] Found pattern:[92:50->92:57]
posCursor:[92:55] posNoWhite:[92:54] Found pattern:[92:54->92:55]
Ppat_construct O:[92:54->92:55]
Completable: CtypedContext(Value[someOtherValue])=O pattern: [RecordField:thirdStuff, Tuple:1, Variant:Five(1)] seenIdents: []
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 95:63
posCursor:[95:63] posNoWhite:[95:62] Found expr:[95:3->95:78]
posCursor:[95:63] posNoWhite:[95:62] Found pattern:[95:29->95:70]
posCursor:[95:63] posNoWhite:[95:62] Found pattern:[95:43->95:69]
posCursor:[95:63] posNoWhite:[95:62] Found pattern:[95:44->95:65]
Ppat_construct Some:[95:44->95:48]
posCursor:[95:63] posNoWhite:[95:62] Found pattern:[95:49->95:64]
posCursor:[95:63] posNoWhite:[95:62] Found pattern:[95:62->95:63]
Ppat_construct O:[95:62->95:63]
Completable: CtypedContext(Value[someOtherValue])=O pattern: [RecordField:fourthStuff, Tuple:0, Variant:Some(0), Polyvariant:WithPayload(0)] seenIdents: []
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 98:67
posCursor:[98:67] posNoWhite:[98:66] Found expr:[98:3->98:78]
posCursor:[98:67] posNoWhite:[98:66] Found pattern:[98:29->98:70]
posCursor:[98:67] posNoWhite:[98:66] Found pattern:[98:42->98:69]
posCursor:[98:67] posNoWhite:[98:66] Found pattern:[98:46->98:68]
Ppat_construct Five:[98:46->98:50]
posCursor:[98:67] posNoWhite:[98:66] Found pattern:[98:50->98:69]
posCursor:[98:67] posNoWhite:[98:66] Found pattern:[98:54->98:67]
posCursor:[98:67] posNoWhite:[98:66] Found pattern:[98:66->98:67]
Ppat_construct T:[98:66->98:67]
Completable: CtypedContext(Value[someOtherValue])=T pattern: [RecordField:thirdStuff, Tuple:1, Variant:Five(1)] seenIdents: [One, Two, T]
[{
    "label": "Three(_)",
    "kind": 4,
    "tags": [],
    "detail": "Three(int)\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 101:43
posCursor:[101:43] posNoWhite:[101:42] Found expr:[101:3->101:52]
posCursor:[101:43] posNoWhite:[101:42] Found pattern:[101:29->101:44]
posCursor:[101:43] posNoWhite:[101:42] Found pattern:[101:42->101:43]
Ppat_construct S:[101:42->101:43]
Completable: CtypedContext(Value[someOtherValue])=S pattern: [RecordField:otherThing] seenIdents: []
[{
    "label": "None",
    "kind": 4,
    "tags": [],
    "detail": "None\n\n",
    "documentation": null
  }, {
    "label": "Some(One)",
    "kind": 4,
    "tags": [],
    "detail": "One\n\n",
    "documentation": null
  }, {
    "label": "Some(Two)",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Some(Three(_))",
    "kind": 4,
    "tags": [],
    "detail": "Three(int)\n\n",
    "documentation": null
  }, {
    "label": "Some(Four(_))",
    "kind": 4,
    "tags": [],
    "detail": "Four(someOtherVariant)\n\n",
    "documentation": null
  }, {
    "label": "Some(Five(_))",
    "kind": 4,
    "tags": [],
    "detail": "Five(someOtherVariant, someVariant)\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 104:39
posCursor:[104:39] posNoWhite:[104:38] Found expr:[104:3->104:48]
posCursor:[104:39] posNoWhite:[104:38] Found pattern:[104:29->104:40]
posCursor:[104:39] posNoWhite:[104:38] Found pattern:[104:38->104:39]
Ppat_construct S:[104:38->104:39]
Completable: CtypedContext(Value[someOtherValue])=S pattern: [RecordField:other2] seenIdents: []
[{
    "label": "None",
    "kind": 4,
    "tags": [],
    "detail": "None\n\n",
    "documentation": null
  }, {
    "label": "Some(_)",
    "kind": 4,
    "tags": [],
    "detail": "Some(_)\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_Records.res 107:64
posCursor:[107:64] posNoWhite:[107:62] Found expr:[107:3->107:75]
posCursor:[107:64] posNoWhite:[107:62] Found pattern:[107:29->107:67]
posCursor:[107:64] posNoWhite:[107:62] Found pattern:[107:42->107:66]
posCursor:[107:64] posNoWhite:[107:62] Found pattern:[107:46->107:65]
Ppat_construct Four:[107:46->107:50]
Completable: CtypedContext(Value[someOtherValue])="" pattern: [RecordField:thirdStuff, Tuple:1, Variant:Four(0)] seenIdents: [TwentyFive]
[{
    "label": "SixtyTwo",
    "kind": 4,
    "tags": [],
    "detail": "SixtyTwo\n\n",
    "documentation": null
  }, {
    "label": "someOtherValue",
    "kind": 12,
    "tags": [],
    "detail": "someRecordWithVariant",
    "documentation": null
  }, {
    "label": "getSomeVal",
    "kind": 12,
    "tags": [],
    "detail": "(~irrelevant: int) => anotherRecord",
    "documentation": null
  }, {
    "label": "someVal",
    "kind": 12,
    "tags": [],
    "detail": "anotherRecord",
    "documentation": null
  }]

